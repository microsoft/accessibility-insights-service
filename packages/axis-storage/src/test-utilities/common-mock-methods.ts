// tslint:disable:no-any no-empty no-unsafe-any no-import-side-effect
import '../node';

import { IMock } from 'typemoq';

export function getPromisableDynamicMock<T>(mock: IMock<T>): IMock<T> {
    // workaround for issue https://github.com/florinn/typemoq/issues/70
    mock.setup((x: any) => x.then).returns(() => undefined);

    return mock;
}
