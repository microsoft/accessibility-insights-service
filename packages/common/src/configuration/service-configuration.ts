// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import * as convict from 'convict';
import * as fs from 'fs';
import { injectable } from 'inversify';
import { isNil } from 'lodash';

export interface CommonRuntimeConfig {
    taskTimeoutInMinutes: number;
    logInConsole: boolean;
    maxQueueSize: number;
}

export interface ScanRunTimeConfig {
    minLastReferenceSeenInDays: number;
    pageRescanIntervalInDays: number;
    failedPageRescanIntervalInHours: number;
    maxScanRetryCount: number;
}

export interface RuntimeConfig {
    commonConfig: CommonRuntimeConfig;
    scanConfig: ScanRunTimeConfig;
}

@injectable()
export class ServiceConfiguration {
    public static readonly profilePath = './runtime-config/runtime-config.json';
    private readonly fileSystem: typeof fs;
    private loadConfigPromise: Promise<convict.Config<RuntimeConfig>>;
    private readonly convictModule: typeof convict;

    constructor(fileSystem: typeof fs = fs, convictModule: typeof convict) {
        this.fileSystem = fileSystem;
        this.convictModule = convictModule;
    }

    public async getConfigValue<K extends keyof RuntimeConfig | null | undefined = undefined>(
        key?: K,
    ): Promise<K extends null | undefined ? RuntimeConfig : RuntimeConfig[K]> {
        const config = await this.getConvictConfig();

        return config.get(key);
    }

    private async getConvictConfig(): Promise<convict.Config<RuntimeConfig>> {
        if (isNil(this.loadConfigPromise)) {
            this.loadConfigPromise = new Promise((resolve, reject) => {
                const config = this.convictModule<RuntimeConfig>(this.getRuntimeConfigSchema());

                this.fileSystem.exists(ServiceConfiguration.profilePath, exists => {
                    if (exists === true) {
                        config.loadFile(ServiceConfiguration.profilePath);
                        config.validate({ allowed: 'strict' });
                    } else {
                        console.log(`Unable to load custom configuration. Using default config  - ${config}`);
                    }
                    resolve(config);
                });
            });
        }

        return this.loadConfigPromise;
    }

    private getRuntimeConfigSchema(): convict.Schema<RuntimeConfig> {
        return {
            commonConfig: {
                logInConsole: {
                    format: 'Boolean',
                    default: true,
                    doc: 'Property to decide if console logging is enabled',
                },
                maxQueueSize: {
                    format: 'int',
                    default: 10,
                    doc: 'Maximum message the queue can have',
                },
                taskTimeoutInMinutes: {
                    format: 'int',
                    default: 3,
                    doc: 'Timeout value after which the task has to be terminated',
                },
            },
            scanConfig: {
                minLastReferenceSeenInDays: {
                    format: 'int',
                    default: 3,
                    doc: 'Minimum days since we last saw a page. Pages older than this time will not be scanned.',
                },
                pageRescanIntervalInDays: {
                    format: 'int',
                    default: 3,
                    doc: 'Minimum days since we last scanned. Pages newer than this time will not be scanned.',
                },
                failedPageRescanIntervalInHours: {
                    format: 'int',
                    default: 3,
                    doc: 'Minimum hours since the last scan failed. Pages newer than this time will not be scanned.',
                },
                maxScanRetryCount: {
                    format: 'int',
                    default: 3,
                    doc: 'Maximum number of retries allowed for a page scan',
                },
            },
        };
    }
}
