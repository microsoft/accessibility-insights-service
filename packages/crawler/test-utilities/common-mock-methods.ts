import { IMock } from 'typemoq';
import { CrawlerRequestOptions } from '../tasks/crawl/hc-crawler-types';

export function getNotAllowedUrls(): string[] {
    return [
        'https://www.bing.com/abc.pdf',
        'https://www.bing.com/abc.js',
        'https://www.bing.com/abc.css',
        'https://www.bing.com/abc.png',
        'https://www.bing.com/abc.jpg',
        'https://www.bing.com/abc.jpeg',
        'https://www.bing.com/abc.gif',
        'https://www.bing.com/abc.json',
        'https://www.bing.com/abc.xml',
        'https://www.bing.com/abc.exe',
        'https://www.bing.com/abc.dmg',
        'https://www.bing.com/abc.ico',
        'https://www.bing.com/abc.txt',
        'https://www.bing.com/abc.zip',
        'https://www.bing.com/abc.war',
        'https://www.bing.com/abc.rar',
        'https://www.bing.com/abc.svg',
    ];
}

export function createCrawlerRequestOptions(requestUrl: string): CrawlerRequestOptions {
    return {
        url: requestUrl,
    };
}

export function getPromisableDynamicMock<T>(mock: IMock<T>): IMock<T> {
    // workaround for issue https://github.com/florinn/typemoq/issues/70

    // tslint:disable-next-line:no-any no-unsafe-any
    mock.setup((x: any) => x.then).returns(() => undefined);

    return mock;
}
