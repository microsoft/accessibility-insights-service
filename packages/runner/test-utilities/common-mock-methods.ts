import { IMock } from 'typemoq';
import { CrawlerRequestOptions, CrawlerRequestResponse, CrawlerResult } from '../src/crawler/hc-crawler-types';

export function getNotAllowedUrls(): string[] {
    return [
        'https://www.bing.com/abc.pdf',
        'https://www.bing.com/abc.js',
        'https://www.bing.com/abc.css',
        'https://www.bing.com/abc.png',
        'https://www.bing.com/abc.jpg',
        'https://www.bing.com/abc.jpeg',
        'https://www.bing.com/abc.gif',
        'https://www.bing.com/abc.json',
        'https://www.bing.com/abc.xml',
        'https://www.bing.com/abc.exe',
        'https://www.bing.com/abc.dmg',
        'https://www.bing.com/abc.ico',
        'https://www.bing.com/abc.txt',
        'https://www.bing.com/abc.zip',
        'https://www.bing.com/abc.war',
        'https://www.bing.com/abc.rar',
        'https://www.bing.com/abc.svg',
    ];
}

export function createCrawlerRequestOptions(requestUrl: string): CrawlerRequestOptions {
    return {
        url: requestUrl,
    };
}

export function getPromisableDynamicMock<T>(mock: IMock<T>): IMock<T> {
    // workaround for issue https://github.com/florinn/typemoq/issues/70

    // tslint:disable-next-line:no-any no-unsafe-any
    mock.setup((x: any) => x.then).returns(() => undefined);

    return mock;
}

export function createCrawlResult(requestUrl: string): CrawlerResult {
    return {
        depth: 1,
        options: {
            url: requestUrl,
            maxDepth: 1,
            priority: 1,
            delay: 0,
            retryCount: 1,
            retryDelay: 0,
            timeout: 0,
            skipDuplicates: true,
            depthPriority: false,
            obeyRobotsTxt: false,
            followSitemapXml: false,
            skipRequestedRedirect: true,
        },
        previousUrl: undefined,
        response: createCrawlerRequestResponse(requestUrl),
        links: [],
    };
}

export function createCrawlerRequestResponse(requestUrl: string): CrawlerRequestResponse {
    return { ok: true, status: 200, url: requestUrl, headers: undefined };
}
