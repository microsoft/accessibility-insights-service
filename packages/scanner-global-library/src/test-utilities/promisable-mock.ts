// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { IMock } from 'typemoq';

export function getPromisableDynamicMock<T>(mock: IMock<T>): IMock<T> {
    // workaround for issue https://github.com/florinn/typemoq/issues/70

    // tslint:disable-next-line:no-any no-unsafe-any
    mock.setup((x: any) => x.then).returns(() => undefined);

    return mock;
}
