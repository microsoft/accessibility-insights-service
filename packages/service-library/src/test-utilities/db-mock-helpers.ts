// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

// tslint:disable: no-unsafe-any no-any align no-constant-condition
import * as cosmos from '@azure/cosmos';
import { CosmosClientWrapper, CosmosOperationResponse } from 'azure-services';
import { HashGenerator } from 'common';
import * as crypto from 'crypto';
import * as _ from 'lodash';
import { ItemType, StorageDocument, Website, WebsitePage, WebsitePageExtra } from 'storage-documents';
import { PageObjectFactory } from '../factories/page-object-factory';

export interface DbContainer {
    dbName: string;
    collectionName: string;
}

const warningMessage = `Warning: To use the Azure Cosmos DB based tests provide Cosmos DB Url and key.
        The Azure Cosmos DB or Azure Cosmos DB emulator https://aka.ms/cosmosdb-emulator (Windows only) can be used.`;
const cosmosDbUrl: string = undefined;
const cosmosDbKey: string = undefined;

export class DbMockHelper {
    public dbContainer: DbContainer;
    public cosmosClient: CosmosClientWrapper;
    private readonly pageFactory = new PageObjectFactory(new HashGenerator());
    private azureCosmosClient: cosmos.CosmosClient;

    public isDbTestSupported(): boolean {
        if (cosmosDbUrl === undefined || cosmosDbKey === undefined) {
            console.log('\x1b[31m', warningMessage);

            return false;
        }

        return true;
    }

    public async init(dbName?: string, collectionName?: string): Promise<boolean> {
        this.dbContainer = {
            dbName: dbName === undefined ? 'test-db' : dbName,
            collectionName: collectionName === undefined ? this.createRandomString('col') : collectionName,
        };

        this.azureCosmosClient = new cosmos.CosmosClient({ endpoint: cosmosDbUrl, auth: { masterKey: cosmosDbKey } });
        this.cosmosClient = new CosmosClientWrapper(() => Promise.resolve(this.azureCosmosClient));

        await this.deleteDbContainer(this.dbContainer);
        await this.createDbContainer(this.dbContainer);

        return true;
    }

    public createPageDocument(options?: {
        label?: string;
        extra?: WebsitePageExtra;
        websiteId?: string;
        baseUrl?: string;
        url?: string;
    }): WebsitePage {
        const websiteId =
            options === undefined || options.websiteId === undefined ? this.createRandomString('websiteId') : options.websiteId;
        const baseUrl = options === undefined || options.baseUrl === undefined ? this.createBaseUrl() : options.baseUrl;
        const url = options === undefined || options.url === undefined ? this.createUrl(baseUrl) : options.url;
        const page = this.pageFactory.createImmutableInstance(websiteId, baseUrl, url);
        (<any>page).label = options === undefined || options.label === undefined ? undefined : options.label;

        if (options.extra !== undefined) {
            _.merge(page, options.extra);
        }

        return page;
    }

    public createWebsiteDocument(options?: {
        label?: string;
        websiteId?: string;
        baseUrl?: string;
        deepScanningEnabled?: boolean;
    }): Website {
        const website = {
            id: this.createRandomString('id'),
            itemType: ItemType.website,
            partitionKey: 'website',
            websiteId: options === undefined || options.websiteId === undefined ? this.createRandomString('websiteId') : options.websiteId,
            name: this.createRandomString('name'),
            baseUrl: options === undefined || options.baseUrl === undefined ? this.createBaseUrl() : options.baseUrl,
            serviceTreeId: this.createRandomString('serviceTreeId'),
        };
        (<any>website).label = options === undefined || options.label === undefined ? undefined : options.label;

        if (options !== undefined && options.deepScanningEnabled) {
            (<any>website).deepScanningEnabled = options.deepScanningEnabled;
        }

        return website;
    }

    public createDocument(itemType: ItemType = ItemType.website, id?: string, partitionKey?: string): StorageDocument {
        return {
            id: id === undefined ? this.createRandomString('id') : id,
            itemType: itemType,
            partitionKey: partitionKey === undefined ? this.createRandomString('pk') : partitionKey,
        };
    }

    public createRandomString(prefix: string = ''): string {
        return `${prefix}-${crypto.randomBytes(5).toString('hex')}`;
    }

    public createBaseUrl(): string {
        return `https://${this.createRandomString()}.localhost/`;
    }

    public createUrl(baseUrl: string): string {
        return `${baseUrl}page-${this.createRandomString()}.html`;
    }

    public async createDbContainer(container: DbContainer): Promise<DbContainer> {
        console.log(`Creating database '${container.dbName}..`);
        const dbResponse = await this.azureCosmosClient.databases.createIfNotExists({ id: container.dbName });
        const db = dbResponse.database;

        await this.waitForDbCreation(container.dbName);

        await db.containers.createIfNotExists(
            {
                id: container.collectionName,
                partitionKey: { paths: [CosmosClientWrapper.PARTITION_KEY_NAME], kind: cosmos.PartitionKind.Hash },
            },
            { offerThroughput: 1000 },
        );

        console.log('.created');

        console.log(`Cosmos container:
    DB: ${container.dbName}
    Collection: ${container.collectionName}`);

        return this.dbContainer;
    }

    public async waitForDbCreation(dbName: string): Promise<void> {
        this.sleep(2000);

        while (true) {
            try {
                await this.azureCosmosClient.database(dbName).read();
                break;
            } catch (error) {
                if ((<cosmos.ErrorResponse>error).code === 404) {
                    continue;
                }

                throw error;
            }
        }
    }

    public async deleteAllDocuments(): Promise<void> {
        const items = await this.cosmosClient.readAllItem(this.dbContainer.dbName, this.dbContainer.collectionName);

        await Promise.all(
            items.item.map(async item => {
                await this.cosmosClient.deleteItem(item.id, this.dbContainer.dbName, this.dbContainer.collectionName, item.partitionKey);
            }),
        );
    }

    public async upsertItem<T>(item: T): Promise<CosmosOperationResponse<T>> {
        return this.cosmosClient.upsertItem(item, this.dbContainer.dbName, this.dbContainer.collectionName, (<any>item).partitionKey);
    }

    public async upsertItems<T>(items: T[]): Promise<void> {
        await Promise.all(
            items.map(async item => {
                await this.upsertItem(item);
            }),
        );
    }

    public getDocumentProjections(documents: any[]): any[] {
        return documents.map(d => {
            return { id: d.id, label: d.label };
        });
    }

    private async deleteDbContainer(container: DbContainer): Promise<void> {
        try {
            console.log(`Deleting database '${container.dbName}..`);
            await this.azureCosmosClient.database(container.dbName).delete();
            this.sleep(2000);

            while (true) {
                try {
                    await this.azureCosmosClient.database(container.dbName).read();
                } catch (error) {
                    if ((<cosmos.ErrorResponse>error).code === 404) {
                        break;
                    }

                    throw error;
                }
            }
            console.log('.deleted');
        } catch (error) {
            if ((<cosmos.ErrorResponse>error).code !== 404) {
                throw error;
            }
        }
    }

    private sleep(time: number): void {
        const stop = new Date().getTime();
        let i = 0;
        while (new Date().getTime() < stop + time) {
            i = i + 1;
        }
    }
}
