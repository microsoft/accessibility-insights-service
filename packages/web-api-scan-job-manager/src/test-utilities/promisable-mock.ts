// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { IMock } from 'typemoq';

export function getPromisableDynamicMock<T>(mock: IMock<T>): IMock<T> {
    // workaround for issue https://github.com/florinn/typemoq/issues/70

    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    mock.setup((x: any) => x.then).returns(() => undefined);

    return mock;
}
