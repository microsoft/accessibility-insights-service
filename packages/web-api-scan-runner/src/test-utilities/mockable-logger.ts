// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { Logger } from 'logger';

export class MockableLogger extends Logger {}
