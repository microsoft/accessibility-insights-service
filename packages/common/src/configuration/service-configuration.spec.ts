// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import 'reflect-metadata';

import * as convict from 'convict';
import * as fs from 'fs';
import { IMock, It, Mock, MockBehavior, Times } from 'typemoq';
import { getPromisableDynamicMock } from '../test-utilities/promisable-mock';
import { RuntimeConfig, ServiceConfiguration } from './service-configuration';

// tslint:disable: non-literal-fs-path no-unsafe-any no-any

describe(ServiceConfiguration, () => {
    let testSubject: ServiceConfiguration;
    let fsMock: IMock<typeof fs>;
    let convictMock: IMock<typeof convict>;
    let configMock: IMock<convict.Config<RuntimeConfig>>;
    let actualSchema: convict.Schema<RuntimeConfig>;

    beforeEach(() => {
        convictMock = Mock.ofInstance(convict);

        configMock = Mock.ofType<convict.Config<RuntimeConfig>>();
        getPromisableDynamicMock(configMock);

        fsMock = Mock.ofInstance(fs, MockBehavior.Strict);
        testSubject = new ServiceConfiguration(fsMock.object, convictMock.object);
    });

    it('verifies dev config', async () => {
        setupVerifiableFileExistsCall(ServiceConfiguration.profilePath, false);
        const keyToFetch = 'commonConfig';
        const expectedConfigValue = 'config value';

        setupVerifiableSchemaSetupCall();
        setupLoadCustomFileConfigCallsNotCalled();
        configMock.setup(c => c.get(keyToFetch)).returns(() => expectedConfigValue as any);

        const actualConfigValue = await testSubject.getConfigValue(keyToFetch);

        expect(actualConfigValue).toBe(expectedConfigValue);
        expect(actualSchema).toMatchSnapshot();

        verifyMocks();
    });

    it('loads config only once', async () => {
        setupVerifiableFileExistsCall(ServiceConfiguration.profilePath, false);
        const keyToFetch = 'commonConfig';
        const expectedConfigValue = 'config value';

        setupVerifiableSchemaSetupCall();
        configMock.setup(c => c.get(keyToFetch)).returns(() => expectedConfigValue as any);

        let actualConfigValue = await testSubject.getConfigValue(keyToFetch);
        expect(actualConfigValue).toBe(expectedConfigValue);

        actualConfigValue = await testSubject.getConfigValue(keyToFetch);
        expect(actualConfigValue).toBe(expectedConfigValue);

        verifyMocks();
    });

    it('verifies custom config', async () => {
        setupVerifiableFileExistsCall(ServiceConfiguration.profilePath, true);
        const keyToFetch = 'commonConfig';
        const expectedConfigValue = 'config value';

        setupVerifiableSchemaSetupCall();
        setupVerifiableLoadCustomFileConfigCalls();

        configMock.setup(c => c.get(keyToFetch)).returns(() => expectedConfigValue as any);

        const actualConfigValue = await testSubject.getConfigValue(keyToFetch);

        expect(actualConfigValue).toBe(expectedConfigValue);
        expect(actualSchema).toMatchSnapshot();

        verifyMocks();
    });

    it('throws on custom config validation failure', async () => {
        setupVerifiableFileExistsCall(ServiceConfiguration.profilePath, true);
        const keyToFetch = 'commonConfig';
        const expectedError = 'validation failed';

        setupVerifiableSchemaSetupCall();
        setupVerifiableCustomConfigLoadCall();
        configMock
            .setup(c => c.validate(It.isValue({ allowed: 'strict' })))
            .returns(() => {
                throw expectedError;
            });

        await expect(testSubject.getConfigValue(keyToFetch)).rejects.toBe(expectedError);

        verifyMocks();
    });

    function verifyMocks(): void {
        configMock.verifyAll();
        convictMock.verifyAll();
        fsMock.verifyAll();
    }

    function setupVerifiableLoadCustomFileConfigCalls(): void {
        setupVerifiableCustomConfigLoadCall();
        configMock.setup(c => c.validate(It.isValue({ allowed: 'strict' }))).verifiable(Times.once());
    }

    function setupVerifiableCustomConfigLoadCall(): void {
        configMock.setup(c => c.loadFile(ServiceConfiguration.profilePath)).verifiable(Times.once());
    }
    function setupLoadCustomFileConfigCallsNotCalled(): void {
        configMock.setup(c => c.loadFile(It.isAny())).verifiable(Times.never());
        configMock.setup(c => c.validate(It.isAny())).verifiable(Times.never());
    }

    function setupVerifiableSchemaSetupCall(): void {
        convictMock
            .setup(c => c<RuntimeConfig>(It.isAny()))
            .callback((s: convict.Schema<RuntimeConfig>) => {
                actualSchema = s;
            })
            .returns(() => configMock.object)
            .verifiable(Times.once());
    }
    function setupVerifiableFileExistsCall(profilePath: string, fileExistsValue: boolean): void {
        fsMock
            .setup(f => f.exists(profilePath, It.isAny()))
            .callback((filePath: string, cb: (exists: boolean) => void) => {
                cb(fileExistsValue);
            })
            .verifiable(Times.once());
    }
});
