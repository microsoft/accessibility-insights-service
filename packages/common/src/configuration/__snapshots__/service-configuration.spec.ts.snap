// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceConfiguration verifies custom config 1`] = `
Object {
  "logInConsole": Object {
    "default": true,
    "doc": "Property to decide if console logging is enabled",
    "format": "Boolean",
  },
  "queueConfig": Object {
    "maxQueueSize": Object {
      "default": 10,
      "doc": "Maximum message the queue can have",
      "format": "int",
    },
  },
  "scanConfig": Object {
    "failedPageRescanIntervalInHours": Object {
      "default": 3,
      "doc": "Minimum hours since the last scan failed. Pages newer than this time will not be scanned.",
      "format": "int",
    },
    "maxScanRetryCount": Object {
      "default": 3,
      "doc": "Maximum number of retries allowed for a page scan",
      "format": "int",
    },
    "minLastReferenceSeenInDays": Object {
      "default": 3,
      "doc": "Minimum days since we last saw a page. Pages older than this time will not be scanned.",
      "format": "int",
    },
    "pageRescanIntervalInDays": Object {
      "default": 3,
      "doc": "Minimum days since we last scanned. Pages newer than this time will not be scanned.",
      "format": "int",
    },
  },
  "taskConfig": Object {
    "taskTimeoutInMinutes": Object {
      "default": 3,
      "doc": "Timeout value after which the task has to be terminated",
      "format": "int",
    },
  },
}
`;

exports[`ServiceConfiguration verifies dev config 1`] = `
Object {
  "logInConsole": Object {
    "default": true,
    "doc": "Property to decide if console logging is enabled",
    "format": "Boolean",
  },
  "queueConfig": Object {
    "maxQueueSize": Object {
      "default": 10,
      "doc": "Maximum message the queue can have",
      "format": "int",
    },
  },
  "scanConfig": Object {
    "failedPageRescanIntervalInHours": Object {
      "default": 3,
      "doc": "Minimum hours since the last scan failed. Pages newer than this time will not be scanned.",
      "format": "int",
    },
    "maxScanRetryCount": Object {
      "default": 3,
      "doc": "Maximum number of retries allowed for a page scan",
      "format": "int",
    },
    "minLastReferenceSeenInDays": Object {
      "default": 3,
      "doc": "Minimum days since we last saw a page. Pages older than this time will not be scanned.",
      "format": "int",
    },
    "pageRescanIntervalInDays": Object {
      "default": 3,
      "doc": "Minimum days since we last scanned. Pages newer than this time will not be scanned.",
      "format": "int",
    },
  },
  "taskConfig": Object {
    "taskTimeoutInMinutes": Object {
      "default": 3,
      "doc": "Timeout value after which the task has to be terminated",
      "format": "int",
    },
  },
}
`;
